### PR TITLE
mutable rate

### PR DESCRIPTION
**Pull request by aybabtme**
@macb @nickvanw @jphines

step 1 in adding support for a `iocontrol.WriterPool` and `iocontrol.ReaderPool`.

### CODE REVIEW

**Comment by jphines** [comment]
could SetRate take a time.duration, as opposed just assuming to be over the span of a second?

File: set_rate_test.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
+package iocontrol
+
+import (
+	"bytes"
+	"github.com/dustin/go-humanize"
```
Comment (macb) [inline set_rate_test.go]: sort your imports?

**Comment by aybabtme** [comment]
@jphines all of `iocontrol` rates are based on Bps, so I think `SetRate` should keep rolling with that.

File: set_rate_test.go (hunk 1 above)
Comment (aybabtme) [inline set_rate_test.go]: why did I invite you here =[

**Comment by jphines** [comment]
kk.

File: set_rate_test.go (hunk 1 above)
Comment (macb) [inline set_rate_test.go]: :trolleybus:

**Comment by nickvanw** [comment]
:+1:

**Comment by macb** [comment]
:+1:

**Pull request merged**